core/chaincode/platforms/golang/hash.go

```diff
@@ -138,7 +135,11 @@ func getCodeFromHTTP(path string) (codegopath string, err error) {
 	//     . more secure
 	//     . as we are not downloading OBC, private, password-protected OBC repo's become non-issue
 
-	env[gopathenvIndex] = "GOPATH=" + codegopath + ":" + origgopath
+	os.Setenv("GOPATH", codegopath+":"+origgopath)
+	// Get a copy of that new env for the go get command
+	env := os.Environ()
+	// and reset GOPATH to its original value
+	os.Setenv("GOPATH", origgopath)
```

Comment:
@muralisrini makes a good point.  It's not clear from reading the golang library spec if this operation would be considered thread/go-routine safe.  If it only impacts the current go-routine, this is probably fine, but I am skeptical this is the case.